### DISCUSSION TITLE
Custom Backed Enum cast

### DISCUSSION THREAD

**Discussion by CristianGiordano** [spatie/laravel-data#886] [Q&A]
Is it possible to create a custom backed enum cast?

I have an incoming webhook payload as :

```
{
    // ....
   "type": "Account", // "Contact" | "Lead" etc
}
```

My enum:

```php
enum WebhookType: string
{
    case Account = 'account';
    case Contact = 'contact';
    case Lead = 'lead';
}
```

I tried extending the `EnumCast` however I think due to the nature of native enums, my `#[WithCast(ToLowercaseEnumCast)]` isn't triggered on my data class.

```php
class ToLowercaseEnumCast extends EnumCast
{
    public function __construct(
        protected ?string $type = null
    ) {
        parent::__construct(gettype($type) === 'string' ? strtolower($type) : $type);
    }
}
```

Is this possible at all? I can just use the same casing from the webhook, but I wanted consistency in my app.

**Comment by rubenvanassche** [comment]
If a property has an attribute cast, it will be the first cast option selected: https://github.com/spatie/laravel-data/blob/8466bd7eb6855f2a59538867d8a60bb685bb9010/src/DataPipes/CastPropertiesDataPipe.php#L61

So you'll have to check why that piece of code isn't triggered.
